src/basilisp/lang/map.py

```diff
@@ -107,6 +125,78 @@ def to_persistent(self) -> "PersistentMap[K, V]":
         return PersistentMap(self._inner.finish())
 
 
+def map_lrepr(  # pylint: disable=too-many-locals
+    entries: Callable[[], Iterable[Tuple[Any, Any]]],
+    start: str,
+    end: str,
+    meta=None,
+    **kwargs: Unpack[PrintSettings],
+) -> str:
+    """Produce a Lisp representation of an associative collection, bookended
+    with the start and end string supplied. The entries argument must be a
+    callable which will produce tuples of key-value pairs.
+
+    If the keyword argument print_namespace_maps is True and all keys
+    are instances of `basilisp.lang.interfaces.INamed` and they share
+    the same namespace, then print the namespace of the keys at the
+    beginning of the map instead of beside the keys.
+
+    The keyword arguments will be passed along to lrepr for the sequence
+    elements.
+
+    """
+    print_level = kwargs["print_level"]
+    if isinstance(print_level, int) and print_level < 1:
+        return SURPASSED_PRINT_LEVEL
+
+    kwargs = process_kwargs(**kwargs)
+
+    def check_same_ns():
+        nses = set()
+        for k, _ in entries():
+            if isinstance(k, INamed):
```

Comment:
```suggestion
            if isinstance(k, INamed) and k.ns is not None:
```

Reply:
We actually need the "None" value to go into the `nses` set, to distinguish the case where we have a namespaced key and a non-namespaced key, in which case we don't want to proceed with lifting the key namespace outside the map.